WordPress/gutenberg
# Improve the 'Show button text labels' preference by using a button wrapper component

Fixes https://github.com/WordPress/gutenberg/issues/61763

So far, this is a Proof Of Concept. Please do not merge.

This PR explores a way to improve the 'Show button text labels' preference by using a wrapper component around the base Button component. The wnew wrapper component lives in the `editor` package, as the 'Show button text labels' preference is specific to the Gutenberg editor. The base Button component stays unchanged.
The wrapper component is responsible to retrieve the preference and add some simple logic to pass the appropriate props to teh Button.

## What?

The 'Show button text labels' preference broke several times as it's inherently fragile and based on CSS pseudo elements.
The `::before` and `::after` pseudo elements are already used for other purposes for the various buttons, tab buttons etc. in teh editor mainly for the focus style, the active style etc. Useing these pseudo elements also to show the button text is fragile and open to CSS conflicts.

## Why?

The 'Show button text labels' preference should be reliable, testable, and not use CSS hacks.

## How?

- Adds a new component in the editor package, _temporarily_ named `PreferencesAwareButton`. Suggestions for better naming welcome.
- Use the new component in the 'document tools' of the top bar, replacing the base Button where appropriate.
- When the preference is enabled:
  - The button `icon` prop is set to `null`.
  - The button `text` prop is set to the `label` value.
  - The button gets a CSS class `is-showing-icon-label` that can be used for style adjustments.

## Testing Instructions




- Go to the post editor > Options > Preferences > Accessibility, and enable the 'Show button text labels' preference.
- Close the preferences modal dialog.
- Observe the buttons in the 'document tools' in the top bar, including the main inserter toggle. Screenshot:

![Screenshot 2024-05-21 at 16 01 32](https://github.com/WordPress/gutenberg/assets/1682452/6bebfcdb-5c7c-4472-87b4-5dd675920eb6)

- Observe there's no visual change compared with the previous implementation.
- Inspect the source with your browser dev tools.
- Observe the buttons don't contains SVG icons. Previously, the icons were hiddnen via CSS.
- Observe the buttons visible text is actual text inside the button elements. Previously, the text was generated via CSS pseudo elements and retrieving the aria-label with `content: attr(aria-label);`.
- Observe the buttons tooltips work as expected. Note: only the buttons with an associated keyboard shortcut are expected to show tooltips.


### Testing Instructions for Keyboard


## Screenshots or screencast
